eventuate-crdt/src/main/scala/com/rbmhtechnology/eventuate/crdt/CRDTService.scala

```diff
@@ -194,15 +194,17 @@ trait CRDTService[A, B] {
         sender() ! GetReply(crdtId, ops.value(crdt))
       case Update(`crdtId`, operation) =>
         ops.prepare(crdt, operation) match {
-          case None =>
-            sender() ! UpdateReply(crdtId, ops.value(crdt))
-          case Some(op) =>
+          case Success(Some(op)) =>
             persist(ValueUpdated(op)) {
               case Success(evt) =>
-                sender() ! UpdateReply(crdtId, ops.value(crdt))
+                sender() ! UpdateReply(crdtId, Success(ops.value(crdt)))
               case Failure(err) =>
                 sender() ! Status.Failure(err)
             }
+          case Success(None) =>
+            sender() ! UpdateReply(crdtId, Success(ops.value(crdt)))
+          case Failure(err) =>
+            sender() ! UpdateReply(crdtId, Failure(err))
```

Comment:
Why not `sender() ! Status.Failure(err)`? This would make the `UpdateReply` changes obsolete.